spec/models/product_spec.rb

```diff
@@ -1,23 +1,55 @@
 describe Spree::Product do
-  let(:gbp_price) { create(:pre_sale_price, amount: 200, currency: 'GBP') }
-  let(:on_sale_variant) { create(:variant, pre_sale_prices: [gbp_price]) }
-  let(:another_on_sale_variant) { create(:variant, pre_sale_prices: [gbp_price]) }
-  let(:not_on_sale_product) { create(:product) }
-  let(:on_sale_product) do
-    create(:product, master: on_sale_variant,
-                     variants: [another_on_sale_variant])
-  end
+  context 'when product on sale' do
+    let(:gbp_price) { create(:pre_sale_price, amount: 200, currency: 'GBP') }
+    let(:on_sale_variant) { create(:variant, pre_sale_prices: [gbp_price]) }
+    let(:another_on_sale_variant) { create(:variant, pre_sale_prices: [gbp_price]) }
+
+    let(:on_sale_product) do
+      create(:product, master: on_sale_variant,
+             variants: [another_on_sale_variant])
+    end
+
+    it 'should be on sale' do
+      expect(on_sale_product.on_sale?('GBP')).to be_truthy
+    end
 
-  context '.pre_sale_price_in' do
-    it 'returns the pre-sale price for the master variant' do
+    it 'should return the pre-sale price for the master variant' do
       expect(on_sale_product.pre_sale_price_in('GBP').amount.to_s).to eq('200.0')
     end
   end
 
-  context '.on_sale?' do
-    it 'returns true if the product has any pre-sale prices set' do
-      expect(on_sale_product.on_sale?).to be_truthy
-      expect(not_on_sale_product.on_sale?).to be_falsy
+  context 'when product not on sale' do
+    let(:not_on_sale_product) { create(:product) }
+
+    it 'should not be on sale' do
+      expect(not_on_sale_product.on_sale?('GBP')).to be_falsy
+    end
+  end
+
+  context 'when the pre-sale price is zero in GBP' do
+    let(:zero_gbp_price) { create(:pre_sale_price, amount: 0, currency: 'GBP') }
+    let(:variant_with_zero_pre_sale_price) { create(:variant, pre_sale_prices: [zero_gbp_price]) }
+    let(:product_with_zero_presale_price) { create(:product, master: variant_with_zero_pre_sale_price) }
+
+    subject { product_with_zero_presale_price.on_sale?('GBP') }
+
+    it { is_expected.to be(false) }
```

Comment:
We're using `be_truthy` and `be_falsy` elsewhere in this spec. Any reason for the change?